metadata-db-app/src/migrations/058_add_provider_id_to_variables_table.clj

```diff
@@ -0,0 +1,26 @@
+(ns migrations.058-add-provider-id-to-variables-table
+  "Adds `provider_id` column to variables table."
+  (:require
+   [clojure.java.jdbc :as j]
+   [cmr.metadata-db.services.provider-validation :as provider-validation]
+   [config.mdb-migrate-helper :as h]))
+
+(defn up
+  "Migrates the database up from version 57 to 58."
+  []
+  (println "migrations.058-add-provider-id-to-variables-table up...")
+  ;; At this point in time, UMM-Vars support is still being developed, so we
+  ;; don't care about what's in the database right now, and the new field
+  ;; can't be null, so:
+  (h/sql "TRUNCATE TABLE METADATA_DB.cmr_variables")
+  (h/sql "TRUNCATE TABLE METADATA_DB.cmr_variable_associations")
+  (h/sql (format (str "ALTER TABLE METADATA_DB.cmr_variables "
+                      "ADD provider_id VARCHAR(%s) NOT NULL")
+                 provider-validation/PROVIDER_ID_MAX_LENGTH))
+  (h/sql "CREATE INDEX variables_p_vn ON METADATA_DB.cmr_variables (provider_id, variable_name)"))
```

Comment:
Thanks for the detailed response!

As for changing the search indices, yeah: this is just the first metadata-db change. We have some more coming. Then we're going to change the routes to include support for provider-id getting passed in the REST-ish APIs (ingest and search). We're not at the point where a user can search variables by provider-id yet, but once we are, we'll be updating the necessary migrations.

Do you see there being a negative impact on the db relationships if we make the current change without updating the search internals?